lm_eval/api/model.py

```diff
@@ -312,7 +317,7 @@ def loglikelihood(
             if context == "":
                 # end of text as context
                 context_enc, continuation_enc = (
-                    [self.eot_token_id],
+                    [self.bos_or_eos_token_id],
```

Comment:
this is in `loglikelihood` (not rolling) - is it ok?

Reply:
I suppose, it is not OK, as you found problem with a task using `loglikelihood_rolling`, and with this line you change all tasks, which use `loglikelihood`

Reply:
It does not matter.
In loglikelihood now the prefix is EOS but there also should be BOS. I don't know how it changes multiple choice scores.

Reply:
i don't understand why `should`, and also why BOS must have priority over EOS, especially if those changes are not thoroughly tested for compatibility with existing state for all the models (current and future ones) from huggingface hub

Reply:
This change is desirable—for the `acc_mutual_info` metric (seldom used) we also calculate unconditional loglikelihoods of answer choices.

These should be conditioned on BOS if it exists, else EOS.

Reply:
May this selected token be reported to logs on a per task basis?